docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,565 @@
+    :width: 300
+
+    Every mesh node is completely independent
+
+    The same array shape and structure used to describe the node positions
+    (pink crosses) in a regular grid (left-hand maps) is equally able to
+    describe **any** position for these nodes (e.g. the right-hand maps),
+    simply by changing the array values. The quadrilateral faces (blue
+    outlines) can therefore be given any quadrilateral shape by re-positioning
+    their constituent nodes.
+
+Faces and volumes can have variable node counts, i.e. different numbers of
+sides. This is achieved by masking the unused 'slots' in the connectivity
+array. See :numref:`ugrid_variable_faces`.
+
+..  _ugrid_variable_faces:
+..  figure:: images/ugrid_variable_faces.svg
+    :alt: Diagram demonstrating mesh faces with variable node counts
+    :align: right
+    :width: 300
+
+    Mesh faces can have different node counts (using masking)
+
+    The 2D connectivity array (blue circles) describes faces by connecting
+    nodes (pink crosses) to make up a face (blue outlines). The faces can use
+    different numbers of nodes by shaping the connectivity array to accommodate
+    the face with the most nodes, then masking unused node 'slots'
+    (black circles) for faces with fewer nodes than the maximum.
+
+Data can be assigned to lines (edges) just as easily as points (nodes) or
+areas (faces). See :numref:`ugrid_edge_data`.
+
+..  _ugrid_edge_data:
+..  figure:: images/ugrid_edge_data.svg
+    :alt: Diagram demonstrating data assigned to mesh edges
+    :align: right
+    :width: 300
+
+    Data can be assigned to mesh edges
+
+    The 2D connectivity array (blue circles) describes edges by connecting 2
+    nodes (pink crosses) to make up an edge (blue lines). Data can be assigned
+    to the edges (orange lines) by matching the indices of the 1D data array
+    (not shown) to the indices in the connectivity array.
+
+.. _ugrid implications:
+
+What does this mean?
+--------------------
+Meshes can represent much more varied spatial arrangements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The highly specific way of recording position (geometry) and shape
+(topology) allows meshes to represent essentially **any** spatial arrangement
+of data. There are therefore many new applications that aren't possible using a
+structured grid, including:
+
+* `The UK Met Office's LFRic cubed-sphere <https://hps.vi4io.org/_media/events/2018/sig-io-uk-adams.pdf>`_
+* `Oceanic model outputs <https://doi.org/10.3390/jmse2010194>`_
+
+.. todo:
+        a third example!
+
+Mesh 'payload' is much larger than with structured grids
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Coordinates are recorded per-node, and connectivities are recorded per-element.
+This is opposed to a structured grid, where a single coordinate value is shared
+by every data point/area along that line.
+
+For example: representing a cubed-sphere using a mesh leads to coordinates and
```

Comment:
I think it's worth noting here that this is true when you are representing a cubed-sphere *with no other coordinates*.

Reply:
3d18be7cdcd8937d45bec9fd83046b2c8dc497d0